pennylane/operation.py

```diff
@@ -1116,23 +1116,21 @@ def compare(self, other):
         >>> ob1.compare(ob2)
         False
         """
-        if isinstance(other, (Tensor, Observable)):
-            return other._obs_data() == self._obs_data()
         if isinstance(other, qml.Hamiltonian):
             return other.compare(self)
+        if isinstance(other, (Tensor, Observable)):
+            return other._obs_data() == self._obs_data()
```

Comment:
This needs to be swapped because now `Hamiltonian` _is_ and observable, so all cases would take the one if statement. Same below.